src/conversions/chrono.rs

```diff
@@ -127,11 +127,12 @@ impl IntoPy<PyObject> for NaiveDate {
 impl FromPyObject<'_> for NaiveDate {
     fn extract(ob: &PyAny) -> PyResult<NaiveDate> {
         let date = <PyDate as PyTryFrom>::try_from(ob)?;
-        Ok(NaiveDate::from_ymd(
+        Ok(NaiveDate::from_ymd_opt(
             date.get_year(),
             date.get_month() as u32,
             date.get_day() as u32,
-        ))
+        )
+        .expect("invalid or out-of-range date"))
```

Comment:
This is the absolute minimum change by basically copying over the code from `chrono`. (Personally, I am not convinced the deprecations in `chrono` are entirely helpful.)

We might want to use this occasion to turn these panics into `PyTypeError` but if I remember the discussion this should never happen as the ranges of `PyDate` and `NaiveData` more or less match up?

Reply:
I generally would argue that panics in a library are only ok if it's almost guaranteed that user code won't trigger them; on that basis I think the `chrono` APIs can be easily misused and so the deprecation seems reasonable (I wonder if they plan to replace them with forms that return errors instead in future).

As for here, I think it's fine to preserve behaviour in this PR however I think given we can't guarantee that these panics will never fire (especially if e.g. `chrono` changes the range it can handle), it's probably good to replace these with `PyValueError` as you suggest. Perhaps leave as an issue for a follow-up PR?